src/storage/mvcc/reader/backward_scanner.rs

```diff
@@ -0,0 +1,758 @@
+    /// load.
+    #[inline]
+    fn handle_last_version(
+        &mut self,
+        some_write: Option<Write>,
+        user_key: &Key,
+    ) -> Result<Option<Value>> {
+        match some_write {
+            None => Ok(None),
+            Some(write) => match write.write_type {
+                WriteType::Put => Ok(Some(self.reverse_load_data_by_write(write, user_key)?)),
+                WriteType::Delete => Ok(None),
+                _ => unreachable!(),
+            },
+        }
+    }
+
+    /// Load the value by the given `some_write`. If value is carried in `some_write`, it will be
+    /// returned directly. Otherwise there will be a default CF look up.
+    ///
+    /// The implementation is similar to `PointGetter::load_data_by_write`.
+    #[inline]
+    fn reverse_load_data_by_write(&mut self, write: Write, user_key: &Key) -> Result<Value> {
+        if self.omit_value {
+            return Ok(vec![]);
+        }
+        match write.short_value {
+            Some(value) => {
+                // Value is carried in `write`.
+                Ok(value)
+            }
+            None => {
+                // Value is in the default CF.
+                self.ensure_default_cursor()?;
+                let value = super::util::near_reverse_load_data_by_write(
+                    &mut self.default_cursor.as_mut().unwrap(),
+                    user_key,
+                    write,
+                    &mut self.statistics,
+                )?;
+                Ok(value)
+            }
+        }
+    }
+
+    /// After `self.reverse_get()`, our write cursor may be pointing to current user key (if we
+    /// found a desired version), or previous user key (if there is no desired version), or
+    /// out of bound.
+    ///
+    /// If it is pointing to current user key, we need to step it until we meet a new
+    /// key. We first try to `prev()` a few times. If still not reaching another user
+    /// key, we `seek_for_prev()`.
+    #[inline]
+    fn move_write_cursor_to_prev_user_key(&mut self, current_user_key: &Key) -> Result<()> {
+        for i in 0..SEEK_BOUND {
+            if i > 0 {
+                self.write_cursor.prev(&mut self.statistics.write);
+            }
+            if !self.write_cursor.valid() {
+                // Key space ended. We are done here.
+                return Ok(());
+            }
+            {
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                if !Key::is_user_key_eq(current_key, current_user_key.encoded().as_slice()) {
+                    // Found another user key. We are done here.
+                    return Ok(());
+                }
+            }
+        }
+
+        // We have not found another user key for now, so we directly `seek_for_prev()`.
+        // After that, we must pointing to another key, or out of bound.
+        self.write_cursor
+            .seek_for_prev(current_user_key, &mut self.statistics.write)?;
```

Comment:
Use internal_seek_for_prev?

Reply:
`seek_for_prev` seems to be valid as well. We are in backward scan mode, and our given key (current_user_key) must be < current cursor key.

Reply:
Actually, using internal_seek_for_prev will let the scanner don't depends on Cursor's behavior, and then if we want to refine Cursor, there will be no fetter.

Reply:
Ok